### ISSUE TITLE
Unexpected Request Timeout errors

### ISSUE THREAD

**Issue by kszafran** [nats-io/nats.go#800]
## Defect

This may be a bug or just me not understanding how to use the client correctly... In any case I'm getting `nats: Request Timeout` errors when using `Fetch` with `nats.Context` on a pull subscription.

#### Versions of `nats.go` and the `nats-server` if one was involved:

nats.go v1.11.0
nats-server 2.3.4

#### OS/Container environment:

macOS 11.5.1

#### Steps or code to reproduce the issue:

Create a stream:
```bash
nats str add TEST --subjects 'TEST.>'
```

Run the following code (I wrote it as a test, but you should be able to run it as `main`, too):
```go
import (
	"context"
	"fmt"
	"github.com/nats-io/nats.go"
	"testing"
)

func TestRequestTimeout(t *testing.T) {
	reproduce()
}

func reproduce() {
	nc, err := nats.Connect(nats.DefaultURL)
	if err != nil {
		panic(err)
	}
	js, err := nc.JetStream()
	if err != nil {
		panic(err)
	}
	sub, err := js.PullSubscribe("TEST.xxx", "testing")
	if err != nil {
		panic(err)
	}

	// normally I'd pass the cancel function to somewhere else
	ctx, _ := context.WithCancel(context.Background())

	for {
		select {
		case <-ctx.Done():
			return
		default:
			msg, err := sub.Fetch(1, nats.Context(ctx))
			if err != nil {
				fmt.Printf("got error: %v\n", err)
			} else {
				msg[0].Ack()
				fmt.Printf("got message: %s\n", msg[0].Data)
			}
		}
	}
}
```

Now wait 5 seconds and publish a message:
```bash
nats pub TEST.yyy 'hello'
```
Wait another 5 seconds and publish another message:
```bash
nats pub TEST.xxx 'hello'
```

#### Expected result:

```
=== RUN   TestRequestTimeout
got message: hello
```

#### Actual result:

```
=== RUN   TestRequestTimeout
got error: nats: Request Timeout
got error: nats: Request Timeout
got message: hello
```

So I'm getting `nats: Request Timeout` error from `Fetch` whenever a message is sent to the stream after `defaultRequestWait`, even if it was sent to a different subject than the one I'm subscribing to.

This does not happen if I pass e.g. `nats.MaxWait(time.Hour)` instead of `nats.Context(ctx)` to `Fetch`. I also found it surprising that I can't pass both `nats.Context` and `nats.MaxWait`, but `Fetch` is still going to use `ttl` even if I pass a context (it just takes the value from `js` configuration).

In my use case I want to pull messages from a topic as long as my application is running. And when the application is shutting down I call `cancel` to interrupt any processes, like consuming from a NATS stream. That's why I'm passing `nats.Context(ctx)` to `Fetch`.

**Comment by kozlovic** [comment]
Just to check: have you tried with the latest from nats.go master branch? If not, could you please?

**Comment by kszafran** [comment]
Sure thing, let me check.

**Comment by kszafran** [comment]
@kozlovic Tested with 9d3a000c8a66 and it seems to work correctly!

Do you have a schedule for the next release, perhaps?

**Comment by kozlovic** [comment]
Yes, I checked myself with your code and it was working. For the release, very very soon, as in maybe this week.
